lib/fs/fakefs.go

```diff
@@ -279,13 +295,18 @@ func (fs *fakefs) Lstat(name string) (FileInfo, error) {
 	if entry == nil {
 		return nil, os.ErrNotExist
 	}
-	return &fakeFileInfo{*entry}, nil
+
+	info := &fakeFileInfo{*entry}
+	if fs.insens {
+		info.name = filepath.Base(name)
```

Comment:
We didn't do the base stuff before. Why do we need it but just in this special case?

Reply:
Because we need `Lstat` to return the name exactly as it was asked. For case-sensitive filesystems it's the only way that can happen anyway, but for case-insensitive ones we explicitly want returned `fakeFileInfo.name` to be whatever was asked for. Could be an `if` here, but since case-sensitive will work anyway, I opted to spare an unnecessary condition. Can add it if you think that would be better.